sycl/doc/extensions/experimental/sycl_ext_oneapi_bfloat16_math_functions.asciidoc

```diff
@@ -103,286 +103,516 @@ then it supports the `bfloat16` math functions described in the next section.
 
 === Math Functions
 
-The following functions are only available when `T` is `bfloat16` or
-`sycl::marray<bfloat16, {N}>`, where `{N}` means any positive value of
-`size_t` type.
+The following functions are only available for scalar `bfloat16`, `marray`,
+and `vec` of type `bfloat16`.
 
 ==== isnan
 
 ```c++
 namespace sycl::ext::oneapi::experimental {
 
-bool isnan(bfloat16 x);
+bool isnan(bfloat16 x);                                                 (1)
 
 template <size_t N>
-sycl::marray<bool, N> isnan(sycl::marray<bfloat16, N> x);
+sycl::marray<bool, N> isnan(sycl::marray<bfloat16, N> x);               (2)
+
+template <typename VecOrSwiz, int N>
+sycl::vec<int16_t, N> isnan(VecOrSwiz<N> x);                            (3)
```

Comment:
Thanks for improving this, but in order to match the existing definition of `sycl::isnan`, the definition should look like:

```
template <typename VecOrSwiz>
/*return type*/ isnan(VecOrSwiz x);
```

I realize that you did not add the `marray` versions, but it would be good to fix those also and unify both cases into a single template, so the total set of `isnan` function overloads for `bfloat16` would be:

```
bool isnan(bfloat16 x);

template <typename NonScalar>
/*return type*/ isnan(NonScalar x);
```

Reply:
Thanks! I've fixed the mismatch in https://github.com/intel/llvm/pull/14002/commits/5f47ea73eeab5f2d55a862d901402e529ee327b8

Reply:
Perfect!  Thanks.